samza-core/src/main/java/org/apache/samza/zk/ProcessorData.java

```diff
@@ -51,4 +54,17 @@ public String getHost() {
   public String getProcessorId() {
     return processorId;
   }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(processorId, host);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) return false;
+    if (getClass() != obj.getClass()) return false;
+    final ProcessorData other = (ProcessorData) obj;
+    return processorId.equals(other.processorId) && host.equals(other.host);
```

Comment:
Can `processorId` be `null`? Can `host` be null? If so, can handle null checks in here as well.